src/controller/cve-id.controller/cve-id.controller.js

```diff
@@ -240,184 +246,262 @@ async function modifyCveId (req, res) {
-    },
-    time: {
-      created: '3000-06-24T21:08:42.937+00:00',
-      modified: '3000-06-24T21:12:46.576+00:00',
-      reserved: '3000-06-24T21:12:46.576+00:00'
-    }
-  },
-  {
-    cve_id: 'CVE-3000-0003',
-    cve_year: 3000,
-    owning_cna: 'orgshortname',
-    state: 'RESERVED',
-    requested_by: {
-      cna: 'orgshortname',
-      user: 'you'
-    },
-    time: {
-      created: '3000-06-24T21:08:42.937+00:00',
-      modified: '3000-06-24T21:12:46.576+00:00',
-      reserved: '3000-06-24T21:12:46.576+00:00'
-    }
-  },
-  {
-    cve_id: 'CVE-3000-0004',
-    cve_year: 3000,
-    owning_cna: 'orgshortname',
-    state: 'RESERVED',
-    requested_by: {
-      cna: 'orgshortname',
-      user: 'you'
-    },
-    time: {
-      created: '3000-06-24T21:08:42.937+00:00',
-      modified: '3000-06-24T21:12:46.576+00:00',
-      reserved: '3000-06-24T21:12:46.576+00:00'
-    }
-  },
-  {
-    cve_id: 'CVE-3000-0005',
-    cve_year: 3000,
-    owning_cna: 'orgshortname',
-    state: 'RESERVED',
-    requested_by: {
-      cna: 'orgshortname',
-      user: 'you'
-    },
-    time: {
-      created: '3000-06-24T21:08:42.937+00:00',
-      modified: '3000-06-24T21:12:46.576+00:00',
-      reserved: '3000-06-24T21:12:46.576+00:00'
-    }
+  if (amount <= 0 && !returned) {
+    returned = true
+    return res.status(404).json(CONSTANTS.INVALID_AMOUNT)
+  }
+
+  if (amount > 1 && batchType === undefined && !returned) {
+    returned = true
+    return res.status(400).json({ message: 'A batch type (i.e. sequential, non-sequential) must be specified when allocating more than one CVE ID.' })
+  }
+
+  let result = await Org.findOne().byShortName(shortName)
+  if (!result && !returned) {
+    returned = true
+    logger.info(shortName + ' CNA does not exist.')
+    return res.status(404).json({ message: shortName + ' CNA does not exist.' })
   }
-]
-
-const nonsequentialIds = [
-  {
-    cve_id: 'CVE-3000-0500',
-    cve_year: 3000,
-    owning_cna: 'orgshortname',
-    state: 'RESERVED',
-    requested_by: {
-      cna: 'orgshortname',
-      user: 'you'
-    },
-    time: {
-      created: '3000-06-24T21:08:42.937+00:00',
-      modified: '3000-06-24T21:12:46.576+00:00',
-      reserved: '3000-06-24T21:12:46.576+00:00'
+
+  let payload
+
+  if (!returned) {
+    payload = {
+      id_quota: result.policies.id_quota
     }
-  },
-  {
-    cve_id: 'CVE-3000-0522',
-    cve_year: 3000,
-    owning_cna: 'orgshortname',
-    state: 'RESERVED',
-    requested_by: {
-      cna: 'orgshortname',
-      user: 'you'
-    },
-    time: {
-      created: '3000-06-24T21:08:42.937+00:00',
-      modified: '3000-06-24T21:12:46.576+00:00',
-      reserved: '3000-06-24T21:12:46.576+00:00'
+
+    result = await CveId.countDocuments({ owning_cna: shortName, cve_year: year, state: 'RESERVED' })
+    payload.total_reserved = result
+    payload.available = (payload.id_quota - payload.total_reserved)
+  }
+
+  if (!returned && payload.available < 0) {
```

Comment:
Just like with the code in org controller, I think we can drop this if statement and that way they get an error message that contains the details of their id quota with the next check.